src/libraries/System.Private.CoreLib/src/System/Globalization/CompareInfo.cs

```diff
@@ -806,23 +577,18 @@ public unsafe bool IsPrefix(ReadOnlySpan<char> source, ReadOnlySpan<char> prefix
 
                 if (options == CompareOptions.Ordinal)
                 {
-                    goto ReturnOrdinal;
+                    return source.StartsWith(prefix);
                 }
-                else if (options == CompareOptions.OrdinalIgnoreCase)
+
+                if (options == CompareOptions.OrdinalIgnoreCase)
                 {
-                    goto ReturnOrdinalIgnoreCase;
+                    return source.StartsWithOrdinalIgnoreCase(prefix);
                 }
-                else
-                {
-                    ThrowCompareOptionsCheckFailed(options);
-                }
-            }
 
-        ReturnOrdinal:
-            return source.StartsWith(prefix);
+                ThrowCompareOptionsCheckFailed(options);
 
-        ReturnOrdinalIgnoreCase:
-            return source.StartsWithOrdinalIgnoreCase(prefix);
+                return false; // make the compiler happy.
```

Comment:
I'm surprised this is needed since `ThrowCompareOptionsCheckFailed` has `DoesNotReturnAttribute`

Reply:
I got a compiler error in some cases so I applied to all cases I touched. I can try again remove it and look what happen.

Reply:
We can do that cleanup in a follow up PR to make the cut for this fix.

Reply:
I tried it one more time andthe compiler still complaining. I'll try to look at this more later as you suggested.